src/main/java/org/prebid/server/bidder/improvedigital/ImprovedigitalBidder.java

```diff
@@ -67,6 +76,46 @@ public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request
         return Result.withValues(httpRequests);
     }
 
+    private ExtUser getAdditionalConsentProvidersUserExt(BidRequest request) {
+        final ExtUser extUser = request.getUser().getExt();
+        if (extUser == null) {
+            return null;
+        }
+
+        final String consentedProviders = ObjectUtils.defaultIfNull(
+                extUser.getConsentedProvidersSettings().getConsentedProviders(), null);
+        if (StringUtils.isBlank(consentedProviders)) {
+            return extUser;
+        }
+
+        final String[] arrayOfSplitString = StringUtils.substringAfter(consentedProviders, "~")
+                .split(REGEX_SPLIT_STRING_BY_DOT);
+
+        return fillExtUser(extUser, arrayOfSplitString);
```

Comment:
If `~` is not present in `consentedProviders` than we no need to modify `user.Ext`:
```
final String consentedProvidersPart = StringUtils.substringAfter(consentedProviders, "~");
if (StringUtils.isEmpty(consentedProvidersPart)) {
     return extUser;
}
        
return fillExtUser(extUser, consentedProvidersPart.split(REGEX_SPLIT_STRING_BY_DOT));
```